### PR TITLE
Update CONFIG.md

### PR DESCRIPTION
**Pull request by leemthompo**
Minor tweaks:

- Puts running connector service on-prem first
- ~Enterprise Search~ —> Search

### CODE REVIEW

**Review by navarone-feekery — APPROVED** [review]
LGTM!

**Comment by leemthompo** [comment]
@navarone-feekery Buildkite is blocking this one:

File: docs/CONFIG.md [hunk 1]
```diff
@@ -33,3 +28,44 @@ Any converted native connector, net-new connector, or customized/modified connec
...
+```yaml
+# Replace the values for `api_key`, `connector_id`, and `service_type` with the values you printed in Kibana.
+
+elasticsearch:
+  api_key: <key1>
+  # Used to write data to .elastic-connectors and .elastic-connectors-sync-jobs
+  # Any connectors without a specific `api_key` value will default to using this key
+connectors:
+  - connector_id: 1234
+    api_key: <key2>
```
Comment (seanstory) [inline docs/CONFIG.md:54]:
```suggestion
    service_type: <service_type>
    api_key: <key2>
```

File: docs/CONFIG.md [hunk 2]
```diff
@@ -33,3 +28,44 @@ Any converted native connector, net-new connector, or customized/modified connec
...
+    # Used to write data to the `search-*` index associated with connector 1234
+    # You may have multiple connectors in your config file!
+  - connector_id: 5678
+  api_key: <key3>
```
Comment (seanstory) [inline docs/CONFIG.md:58]:
```suggestion
    service_type: <service_type>
    api_key: <key3>
```

File: docs/CONFIG.md [hunk 3]
```diff
@@ -33,3 +28,44 @@ Any converted native connector, net-new connector, or customized/modified connec
...
+  # Used to write data to the `search-*` index associated with connector 5678
+  - connector_id: abcd
```
Comment (seanstory) [inline docs/CONFIG.md:60]:
```suggestion
  - connector_id: abcd
    service_type: <service_type>
```

**Review by seanstory — APPROVED** [review]
Thanks! Several suggestion comments, just to make the config valid (also one `api_key` wasn't indented far enough) but the content is great. Thanks again!

**Comment by seanstory** [comment]
whoops, didn't realize auto-merge was on. :/

**Pull request merged**